### PR TITLE
Restore support for some SIMD arithmetic intrinsics

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

Changes:
 * Restores `simd_add`, `simd_sub` and `simd_mul` intrinsics.
 * `codegen_simd_op_with_overflow` is now a function instead of a macro.
 * Converts the overflow test into an `expected` one.

### Resolved issues:

Towards #1148 

### Call-outs:

 * Tried to do #1483 for these but it would force us to unfold the SIMD operations ourselves.
 * Some of these checks are quite expensive, so I'm testing the shortest bitwidth only.

### Testing:

* How is this change tested? Existing tests + modified tests.

* Is this a refactor change? Yes, mostly.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 1]
```diff
@@ -1432,6 +1417,44 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_expr_to_place(p, e)
     }
 
+    // Intrinsics which encode a SIMD arithmetic operation with overflow check.
+    // We expand the overflow check because CBMC overflow operations don't accept array as
+    // argument.
+    fn codegen_simd_op_with_overflow<F: FnOnce(Expr, Expr) -> Expr, G: Fn(Expr, Expr) -> Expr>(
```
Comment (tedinski) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:1423]: Doc comments have 3 ///

But also: love love love seeing macros get converted to functions here! 😄

File: kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs [hunk 2]
```diff
@@ -1432,6 +1417,44 @@ impl<'tcx> GotocCtx<'tcx> {
...
+
+        let mut check = Expr::bool_false();
+        for i in 0..a_size {
+            // create expression
+            let index = Expr::int_constant(i, Type::ssize_t());
+            let v_a = a.clone().index_array(index.clone());
+            let v_b = b.clone().index_array(index);
+            check = check.or(overflow_fun(v_a, v_b));
+        }
+        let check_stmt = self.codegen_assert(
```
Comment (tedinski) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs:1447]: Should be `codegen_assert_assume`.

Comment (adpaco-aws) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs]: Done!

**Comment by adpaco-aws** [comment]
Thanks @tedinski , just addressed your comments.

**Review by tedinski — APPROVED** [review]

**Pull request merged**